base-containers/base/bin/_run_student_intern

```diff
@@ -79,17 +76,33 @@ print("Chdir")
 os.chdir(start_cmd["working_dir"])
 print("Popen")
 print(fds)
-p = subprocess.Popen(shlex.split(start_cmd["command"]), preexec_fn=setlimits, stdin=fds[0], stdout=fds[1], stderr=fds[2])
 
+set_limits = (lambda: set_limits_user(user))  # To know if the command should be executed as root or worker
+if start_cmd["command"] is None or start_cmd["command"] == "":  # Avoid to ron an empty command
+    start_cmd["command"] = "echo 'student_container started' "
+
+# Run the command
+p = subprocess.Popen(shlex.split(start_cmd["command"]), preexec_fn=set_limits, stdin=fds[0], stdout=fds[1], stderr=fds[2])
 # Handle the signals
 print("Signal thread")
 thread = threading.Thread(target=lambda: handle_signals(p, client), daemon=True)
 thread.start()
-
 # Wait until the subprocess exit
 print("Waiting for process to end")
 retval = p.wait()
 
-# Exit with the same return value than the process (it will be given to docker and given back to run_student)
-print("Exiting")
-exit(retval)
+if start_cmd["ssh"]:
+    # Start ssh server
+    ssh_user, password = start_ssh_server(user)
+    # Send ssh information
+    ssh_user_value = 0 if ssh_user == "worker" else 1  # 0 for worker, 1 for root
+    message = {"type": "ssh_student", "ssh_user": ssh_user_value, "password": password}
+    assert (sys.getsizeof(message)) == 240  # Ensure a constant size of 240 bytes
```

Comment:
It is highly probable that ``sys.getsizeof`` is architecture-dependent. There may be a more appropriate msgpack API to compute this. I'm not even sure this check would be useful at this stage.

Reply:
Indeed, the way I checked its size was incorrect.
I was way better to tansform the message into bytes with msgpack before checking its len.